web/src/settings_users.js

```diff
@@ -631,3 +638,38 @@ export function set_up_bots() {
         settings_bots.add_a_new_bot();
     });
 }
+
+export function fetch_presence_user_setting(render_table) {
+    if (page_params.is_spectator) {
+        render_table();
+        return;
+    }
+
+    channel.post({
+        url: "/json/users/me/presence",
+        data: {
+            status: compute_active_status(),
+            ping_only: false,
+            last_update_id: -1,
+            history_limit_days: 9999,
+        },
+        success(response) {
+            const data = post_presence_response_schema.parse(response);
+
+            if (data.presences) {
+                user_setting_presence_info.clear();
+
+                for (const [user_id_str, presence_data] of Object.entries(data.presences)) {
+                    const user_id = Number.parseInt(user_id_str, 10);
+                    user_setting_presence_info.set(user_id, {
+                        active_timestamp: presence_data.active_timestamp,
+                        idle_timestamp: presence_data.idle_timestamp,
+                    });
```

Comment:
There's a few things that could be improved with this logic:
- We should only ever need to do this sort of full fetch once, not every time we load the "Users" panel. Likely some sort of state variable to track if we've fetched this would be appropriate.
- Ideally, we'd be able to use these data in other UI elements that display presence, not just here. I think we can just store these data via `presence.set_info(data.presences, data.server_timestamp, data.last_update_id)`, with a comment explaining that the next regular default presence checkin with the server should naturally pick up from here.
- I think it might be nice to have the "Users" table load immediately and just show loading indicators in the relevant column for any users for whom we don't have presence data, and trigger a rerender when this request returns. Is that an easy restructuring?

Reply:
@timabbott, from your comment, I understand that I may not need to introduce an extra variable `user_setting_presence_info` here. Instead, I should fetch the complete data on the initial load of the application and then continue updating it through regular `update-user-presence` calls, using this data wherever we need to display presence information.

Am I understanding this correctly?

> Is that an easy restructuring?

I'm not entirely sure, but I’ll give it a try and see how it goes. 

Thank you!

Reply:
> * We should only ever need to do this sort of full fetch once, not every time we load the "Users" panel. Likely some sort of state variable to track if we've fetched this would be appropriate.

Introduced a presence_data_fetched flag to keep track of whether this data has been fetched.

> * Ideally, we'd be able to use these data in other UI elements that display presence, not just here. I think we can just store these data via `presence.set_info(data.presences, data.server_timestamp, data.last_update_id)`, with a comment explaining that the next regular default presence checkin with the server should naturally pick up from here.

Implemented this approach and removed the extra variable I initially introduced.

> * I think it might be nice to have the "Users" table load immediately and just show loading indicators in the relevant column for any users for whom we don't have presence data, and trigger a rerender when this request returns. Is that an easy restructuring?

 I found this restructuring challenging. I attempted it but was not successful.

@timabbott, could you please review and let me know if I’ve correctly addressed the first two points?
Thanks!